polyfill/lib/ecmascript.mjs

```diff
@@ -495,6 +516,33 @@ export const ES = ObjectAssign({}, ES2019, {
     options = ES.NormalizeOptionsObject(options);
     return ES.GetOption(options, 'largestUnit', [...allowed], fallback);
   },
+  ToSmallestTemporalUnit: (options, disallowedStrings = []) => {
+    const singular = {
+      days: 'day',
+      hours: 'hour',
+      minutes: 'minute',
+      seconds: 'second',
+      milliseconds: 'millisecond',
+      microseconds: 'microsecond',
+      nanoseconds: 'nanosecond'
+    };
+    const allowed = new Set(['day', 'hour', 'minute', 'second', 'millisecond', 'microsecond', 'nanosecond']);
+    for (const s of disallowedStrings) {
+      allowed.delete(s);
+    }
+    const allowedValues = [...allowed];
+    options = ES.NormalizeOptionsObject(options);
+    let value = options.smallestUnit;
+    if (value !== undefined) {
+      value = ES.ToString(value);
+      if (value in singular) value = singular[value];
```

Comment:
```suggestion
      if (Object.hasOwnProperty(singular, value)) value = singular[value];
```
In particular, this avoids getting properties off `Object.protoype`. Or use a `Map`?

Reply:
I think you mean this:
```suggestion
      if (Object.prototype.hasOwnProperty.call(singular, value)) value = singular[value];
```